modules/touch/touchmodule.cpp

```diff
@@ -28,44 +28,98 @@
+#include <openspace/engine/openspaceengine.h>
 #include <openspace/engine/windowdelegate.h>
 #include <openspace/interaction/interactionmonitor.h>
 #include <openspace/navigation/navigationhandler.h>
+#include <openspace/rendering/renderable.h>
+#include <openspace/util/factorymanager.h>
+#include <algorithm>
 
 using namespace TUIO;
 
 namespace {
-    constexpr openspace::properties::Property::PropertyInfo TouchActiveInfo = {
-        "TouchActive",
-        "True if we want to use touch input as 3D navigation",
-        "Use this if we want to turn on or off Touch input navigation. "
-        "Disabling this will reset all current touch inputs to the navigation."
+    constexpr std::string_view _loggerCat = "TouchModule";
+
+    constexpr openspace::properties::Property::PropertyInfo EnableTouchInfo = {
+        "EnableTouchInteraction",
+        "Enable Touch Interaction",
+        "Use this property to turn on/off touch input navigation in the 3D scene. "
+        "Disabling will reset all current touch inputs to the navigation."
+    };
+
+    constexpr openspace::properties::Property::PropertyInfo EventsInfo = {
+        "DetectedTouchEvent",
+        "Detected Touch Event",
+        "True when there is an active touch event",
+        openspace::properties::Property::Visibility::Hidden
+    };
+
+    constexpr openspace::properties::Property::PropertyInfo
+        DefaultDirectTouchRenderableTypesInfo =
+    {
+        "DefaultDirectTouchRenderableTypes",
+        "Default Direct Touch Renderable Types",
+        "A list of renderable types that will automatically use the \'direct "
+        "manipulation\' scheme when interacted with, keeping the finger on a static "
+        "position on the interaction sphere of the object when touching. Good for "
+        "relatively spherical objects.",
+        openspace::properties::Property::Visibility::AdvancedUser
     };
 }
 namespace openspace {
 
 TouchModule::TouchModule()
     : OpenSpaceModule("Touch")
-    , _touchActive(TouchActiveInfo, true)
+    , _touchIsEnabled(EnableTouchInfo, true)
+    , _hasActiveTouchEvent(EventsInfo, false)
+    , _defaultDirectTouchRenderableTypes(DefaultDirectTouchRenderableTypesInfo)
 {
     addPropertySubOwner(_touch);
     addPropertySubOwner(_markers);
-    addProperty(_touchActive);
-    _touchActive.onChange([&] {
+    addProperty(_touchIsEnabled);
+    _touchIsEnabled.onChange([&]() {
         _touch.resetAfterInput();
         _lastTouchInputs.clear();
     });
+
+    _hasActiveTouchEvent.setReadOnly(true);
+    addProperty(_hasActiveTouchEvent);
+
+    _defaultDirectTouchRenderableTypes.onChange([&]() {
+        _sortedDefaultRenderableTypes.clear();
+        for (const std::string& s : _defaultDirectTouchRenderableTypes.value()) {
+            ghoul::TemplateFactory<Renderable>* fRenderable =
+                FactoryManager::ref().factory<Renderable>();
+
+            if (!fRenderable->hasClass(s)) {
+                LWARNING(fmt::format(
```

Comment:
Should we maybe lower the level of this logging? Thinking that if you are close to a node of unsupported type this could clutter the log. Or maybe it will only warn once?

Reply:
It only warns when setting the property value, which people would not really do that often during runtime :)